app/models/manifest_uploader.rb

```diff
@@ -35,20 +35,18 @@ def location_barcodes
   end
 
   def locations
-    @locations ||= {}.tap do |h|
-      location_barcodes.each do |barcode|
-        h[barcode] = Location.find_by(barcode: barcode)
-      end
-    end
+    @locations ||= Location.where(barcode: location_barcodes)
+                           .include_for_labware_receipt
+                           .index_by(&:barcode)
   end
 
-  def empty_locations
-    @empty_locations ||= locations.select { |_k, v| v.nil? }
+  def missing_locations
+    @missing_locations ||= location_barcodes.reject { |barcode| locations.key?(barcode) }
   end
```

Comment:
Missing barcodes are no longer present in the hash, so we build up the list by going over the original array instead.

Reply:
Oh, and I renamed the method, as empty could imply locations not containing any labware.